opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -26,21 +28,46 @@
 class LabelSet(metrics_api.LabelSet):
     """See `opentelemetry.metrics.LabelSet`."""
 
-    def __init__(self, labels: Dict[str, str] = None):
+    def __init__(self, labels: Dict[str, str] = None, encoded=""):
```

Comment:
I'm not sure if there is a way to enforce the syncing of `labels` to the encoding without giving up the optimization of not having to calculate the encoding value everytime. To not have to calculate the encoding value, it should be calculated once and then stored in the `LabelSet`. However, since it is now stored as a variable, it or the `labels` can theoretically be changed and there's nothing that can enforce that. 

I think what we can do is to have the `encoded` value calculated on construction, and then have the `__hash__` and `__eq__` be a simple comparison. Along side this, we can make the `labels` and `encoded` protected fields that are tuples, to indicate they should not be changed and are to be used purely for comparisons.